src/bundle/Templating/Twig/Functions/AbstractFunction.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClientBundle\Templating\Twig\Functions;
+
+use Twig\Environment as TwigEnvironment;
+
+abstract class AbstractFunction
```

Comment:
Consider changing to `trait TwigAware` and use `ServiceSubscriberTrait & ServiceSubscriberInterface`.

Reply:
As we discussed, this approach will not change